<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Interactive visualisation for microbiome data â€¢ iSEEtree</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><link href="extra.css" rel="stylesheet">
<meta property="og:title" content="Interactive visualisation for microbiome data">
<meta property="og:description" content="iSEEtree is an extension of iSEE for the TreeSummarizedExperiment. It leverages the funcionality from the miaViz package for microbiome data visualisation to create panels that are specific for TreeSummarizedExperiment objects. Not surprisingly, it also depends on the generic panels from iSEE.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">iSEEtree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="articles/iSEEtree.html">Get started</a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microbiome/iSEEtree/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="iseetree">iSEEtree<a class="anchor" aria-label="anchor" href="#iseetree"></a>
</h1></div>
<p><a href="https://github.com/microbiome/iSEEtree/issues" class="external-link"><img src="https://img.shields.io/github/issues/microbiome/iSEEtree" alt="issues"></a> <a href="https://github.com/microbiome/iSEEtree/pulls" class="external-link"><img src="https://img.shields.io/github/issues-pr/microbiome/iSEEtree" alt="pulls"></a> <a href="https://github.com/microbiome/iSEEtree/actions" class="external-link"><img src="https://github.com/microbiome/iSEEtree/workflows/rworkflows/badge.svg" alt="R-CMD-check"></a> <a href="https://app.codecov.io/gh/microbiome/iSEEtree?branch=devel" class="external-link"><img src="https://codecov.io/gh/microbiome/iSEEtree/branch/devel/graph/badge.svg" alt="codecov"></a> <a href="https://www.codefactor.io/repository/github/microbiome/iseetree" class="external-link"><img src="https://www.codefactor.io/repository/github/microbiome/iseetree/badge" alt="codefactor"></a></p>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>iSEEtree is a Bioconductor package for the interactive visualisation of microbiome data stored in a TreeSummarizedExperiment (TreeSE) container. On the one side, it leverages and extends the graphics of the iSEE package, which is designed for the generic SummarizedExperiment class. On the other side, it employs the statistical and visual tools for microbiome data science provided by the mia family of packages. Thus, iSEE and mia represent the two building blocks of iSEEtree. Detailed introductory material on these two frameworks is available in the <a href="https://isee.github.io/" class="external-link">iSEE-verse website</a> and the <a href="https://microbiome.github.io/OMA/docs/devel/" class="external-link">OMA Bioconductor book</a>, respectively.</p>
<p>iSEEtree is meant for new and experienced users alike, who desire to create and interact with several graphics for microbiome data, without an in-depth knowledge of the underlying mia functionality. Current microbiome-specific panels include phylogenetic trees, ordination plots and compositional plots, which can be further explored below in this article. Other more generic panels are also reused from the iSEE package and can be experimented in <a href="https://isee.github.io/iSEE/articles/basic.html" class="external-link">this article</a>.</p>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>The release version of iSEEtree can be installed from Bioconductor as follows:</p>
<pre><code><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"iSEEtree"</span><span class="op">)</span></span></code></pre>
<p>Contributors or users interested in the latest functionality can install the devel version of iSEEtree as follows:</p>
<pre><code><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># The following initializes usage of Bioc devel</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span>version<span class="op">=</span><span class="st">'devel'</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"iSEEtree"</span><span class="op">)</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h2>
<p>The basic functionality of iSEEtree can be explored as follows:</p>
<pre><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/iSEEtree" class="external-link">iSEEtree</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import TreeSE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Tengeler2020"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">Tengeler2020</span></span>
<span></span>
<span><span class="co"># Add relabundance assay</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add reduced dimensions</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span><span class="va">tse</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Launch iSEE</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="reference/iSEE.html">iSEE</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="panels">Panels<a class="anchor" aria-label="anchor" href="#panels"></a>
</h2>
<p>iSEEtree derives its microbiome-related visualisation methods from the <a href="https://microbiome.github.io/miaViz/" class="external-link">miaViz</a> package, which is code-based and requires basic knowledge of R programming and microbiome data structures. The following panels represent an easy-to-use interactive version of the miaViz plotting functions:</p>
<ul>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/AbundanceDensityPlot.html" class="external-link">AbundanceDensityPlot</a>: a density plot of the top features, where every line is a feature and the x axis shows its abundance for different samples. Its interpretation is explained in the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/12_quality_control.html" class="external-link">Exploration</a>.</li>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/AbundancePlot.html" class="external-link">AbundancePlot</a>: a compositional barplot of the samples, where every bar is a sample composed by different features in different colours. Its interpretation is explained in the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/21_microbiome_community.html" class="external-link">Community Composition</a>.</li>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/RDAPlot.html" class="external-link">RDAPlot</a>: an supervised ordination plot of the samples, where every dot is a sample on a reduced dimensional space and every arrow reflects the contribution of a sample variable. Its interpretation is explained in the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/20_beta_diversity.html" class="external-link">Community Similarity</a>.</li>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/RowTreePlot.html" class="external-link">RowTreePlot</a>: a phylogenetic tree of the features, where every tip is a feature and two neighbouring tips are more closely related than two far apart from each other.</li>
</ul>
<p>By default, the iSEEtree layout also includes the following panels inherited by iSEE:</p>
<ul>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#row-data-tables" class="external-link">RowDataTable</a></li>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#column-data-tables" class="external-link">ColumnDataTable</a></li>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#reduced-dimension-plots" class="external-link">ReducedDimensionPlot</a></li>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#heat-maps" class="external-link">ComplexHeatmapPlot</a></li>
</ul>
<p>The <a href="https://isee.github.io/iSEE/articles/basic.html#coldataplot" class="external-link">ColumnDataPlot</a> could also prove useful for the visualisation of column variables such as alpha diversity indices. Its interpretation is explained in the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/14_alpha_diversity.html" class="external-link">Community Diversity</a>.</p>
</div>
<div class="section level2">
<h2 id="code-of-conduct">Code of Conduct<a class="anchor" aria-label="anchor" href="#code-of-conduct"></a>
</h2>
<p>Please note that the iSEEtree project is released with a <a href="https://bioconductor.org/about/code-of-conduct/" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms. Contributions are welcome in the form of feedback, issues and pull requests. You can find the contributor guidelines of the miaverse <a href="https://github.com/microbiome/mia/blob/devel/CONTRIBUTING.md" class="external-link">here</a>.</p>
</div>
<div class="section level2">
<h2 id="acknowledgements">Acknowledgements<a class="anchor" aria-label="anchor" href="#acknowledgements"></a>
</h2>
<p>Please note that iSEEtree was only made possible thanks to many other R and bioinformatics software authors, which are cited either in the vignettes and/or the paper(s) describing this package. In particular, iSEEtree implements the <a href="https://microbiome.github.io/miaViz/" class="external-link"><em>miaViz</em></a> package for microbiome data visualisation to create panels that are specific for TreeSummarizedExperiment objects. Not surprisingly, it also depends on the generic panels from iSEE.</p>
<p>This package was developed using the following resources:</p>
<ul>
<li>
<a href="https://cran.r-project.org/web/packages/usethis/" class="external-link"><em>usethis</em></a> to generate an initial template.</li>
<li>Continuous code testing is performed on <a href="https://github.com/features/actions" class="external-link">GitHub actions</a> and include R CMD check, <a href="https://bioconductor.org/packages/3.16/bioc/html/BiocCheck.html" class="external-link"><em>BiocCheck</em></a> and testthat.</li>
<li>Code coverage assessment is possible thanks to <a href="https://app.codecov.io/gh/" class="external-link">codecov</a>.</li>
<li>The documentation website is automatically updated thanks to <a href="https://cran.r-project.org/web/packages/pkgdown/" class="external-link"><em>pkgdown</em></a>.</li>
<li>The documentation is formatted thanks to <a href="https://cran.r-project.org/web/packages/devtools/" class="external-link"><em>devtools</em></a> and <a href="https://cran.r-project.org/web/packages/roxygen2/" class="external-link"><em>roxygen2</em></a>.</li>
<li>All the actions above are made reproducible by <a href="https://neurogenomics.github.io/rworkflows/" class="external-link"><em>rworkflows</em></a>
</li>
</ul>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/microbiome/iSEEtree/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/microbiome/iSEEtree/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/Artistic-2.0" class="external-link">Artistic-2.0</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing iSEEtree</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Giulio Benedetti <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-8732-7692" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Giulio Benedetti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

      </footer>
</div>






  </body>
</html>
