<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>iSEEtree: interactive exploration of microbiome data • iSEEtree</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="iSEEtree: interactive exploration of microbiome data">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEtree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/iSEEtree.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metagenomic_data.html">Metagenomic Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microbiome/iSEEtree/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>iSEEtree: interactive exploration of microbiome
data</h1>
                        <h4 data-toc-skip class="author">Giulio
Benedetti</h4>
            <address class="author_afil">
      University of
Turku<br><a class="author_email" href="mailto:#"></a><a href="mailto:giulio.benedetti@utu.fi" class="email">giulio.benedetti@utu.fi</a>
      </address>
                  
            <h4 data-toc-skip class="date">3 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/microbiome/iSEEtree/blob/tutorial/vignettes/iSEEtree.Rmd" class="external-link"><code>vignettes/iSEEtree.Rmd</code></a></small>
      <div class="hidden name"><code>iSEEtree.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<div class="section level3">
<h3 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h3>
<p>iSEEtree is a Bioconductor package for the interactive visualisation
of microbiome data stored in a TreeSummarizedExperiment (TreeSE)
container. On the one side, it leverages and extends the graphics of the
iSEE package, which is designed for the generic SummarizedExperiment
class. On the other side, it employs the statistical and visual tools
for microbiome data science provided by the mia family of packages.
Thus, iSEE and mia represent the two building blocks of iSEEtree.
Detailed introductory material on these two frameworks is available in
the <a href="https://isee.github.io/" class="external-link">iSEE-verse website</a> and the <a href="https://microbiome.github.io/OMA/docs/devel/" class="external-link">OMA Bioconductor
book</a>, respectively.</p>
<p>iSEEtree is meant for new and experienced users alike, who desire to
create and interact with several graphics for microbiome data, without
the need for an in-depth knowledge of the underlying mia functionality.
Current microbiome-specific panels include phylogenetic trees,
ordination plots and compositional plots, which can be further explored
below in this article. Other more generic panels are also reused from
the iSEE package and can be experimented in <a href="https://isee.github.io/iSEE/articles/basic.html" class="external-link">this
article</a>.</p>
</div>
<div class="section level3">
<h3 id="panels">Panels<a class="anchor" aria-label="anchor" href="#panels"></a>
</h3>
<p>iSEEtree derives its microbiome-related visualisation methods from
the <a href="https://microbiome.github.io/miaViz/" class="external-link">miaViz</a> package,
which is code-based and requires basic knowledge of R programming and
microbiome data structures. The following panels represent an
easy-to-use interactive version of the miaViz plotting functions:</p>
<ul>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/AbundanceDensityPlot.html" class="external-link">AbundanceDensityPlot</a>:
a density plot of the top features, where every line is a feature and
the x axis shows its abundance for different samples. Its interpretation
is explained in the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/12_quality_control.html" class="external-link">Exploration</a>.</li>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/AbundancePlot.html" class="external-link">AbundancePlot</a>:
a compositional barplot of the samples, where every bar is a sample
composed by different features in different colours. Its interpretation
is explained in the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/21_microbiome_community.html" class="external-link">Community
Composition</a>.</li>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/RDAPlot.html" class="external-link">RDAPlot</a>:
an supervised ordination plot of the samples, where every dot is a
sample on a reduced dimensional space and every arrow reflects the
contribution of a sample variable. Its interpretation is explained in
the OMA chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/20_beta_diversity.html" class="external-link">Community
Similarity</a>.</li>
<li>
<a href="https://microbiome.github.io/iSEEtree/reference/RowTreePlot.html" class="external-link">RowTreePlot</a>:
a phylogenetic tree of the features, where every tip is a feature and
two neighbouring tips are more closely related than two far apart from
each other.</li>
</ul>
<p>By default, the iSEEtree layout also includes the following panels
inherited by iSEE:</p>
<ul>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#row-data-tables" class="external-link">RowDataTable</a></li>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#column-data-tables" class="external-link">ColumnDataTable</a></li>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#reduced-dimension-plots" class="external-link">ReducedDimensionPlot</a></li>
<li><a href="https://isee.github.io/iSEE/articles/basic.html#heat-maps" class="external-link">ComplexHeatmapPlot</a></li>
</ul>
<p>The <a href="https://isee.github.io/iSEE/articles/basic.html#coldataplot" class="external-link">ColumnDataPlot</a>
could also prove useful for the visualisation of column variables such
as alpha diversity indices. Its interpretation is explained in the OMA
chapter on <a href="https://microbiome.github.io/OMA/docs/devel/pages/14_alpha_diversity.html" class="external-link">Community
Diversity</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="tutorial">Tutorial<a class="anchor" aria-label="anchor" href="#tutorial"></a>
</h2>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>R is an open-source statistical environment which can be easily
modified to enhance its functionality via packages. iSEEtree is an R
package available on <a href="https://bioconductor.org/" class="external-link">Bioconductor</a>. R can be installed on
any operating system from <a href="https://cran.r-project.org/" class="external-link">CRAN</a>
after which you can install iSEEtree by using the following commands in
your R session:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"iSEEtree"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="example">Example<a class="anchor" aria-label="anchor" href="#example"></a>
</h3>
<p>The panels described above can be generated for a model TreeSE object
in the following example:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/iSEEtree" class="external-link">iSEEtree</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Import TreeSE</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"Tengeler2020"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">Tengeler2020</span></span>
<span></span>
<span><span class="co"># Add relabundance assay</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Add reduced dimensions</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span><span class="va">tse</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Launch iSEE</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/iSEE.html">iSEE</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="screenshots%2Fget_started.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="resources">Resources<a class="anchor" aria-label="anchor" href="#resources"></a>
</h2>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>We hope that iSEEtree will be useful for your research. Please use
the following information to cite the package and the overall approach.
Thank you!</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Citation info</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/citation.html" class="external-link">citation</a></span><span class="op">(</span><span class="st">"iSEEtree"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in citation("iSEEtree"): could not determine year for 'iSEEtree' from</span></span>
<span><span class="co">#&gt; package DESCRIPTION file</span></span>
<span><span class="co">#&gt; To cite package 'iSEEtree' in publications use:</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   Benedetti G, Lahti L (????). _iSEEtree: Interactive visualisation for</span></span>
<span><span class="co">#&gt;   microbiome data_. R package version 0.99.7,</span></span>
<span><span class="co">#&gt;   &lt;https://github.com/microbiome/iSEEtree&gt;.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; A BibTeX entry for LaTeX users is</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;   @Manual{,</span></span>
<span><span class="co">#&gt;     title = {iSEEtree: Interactive visualisation for microbiome data},</span></span>
<span><span class="co">#&gt;     author = {Giulio Benedetti and Leo Lahti},</span></span>
<span><span class="co">#&gt;     note = {R package version 0.99.7},</span></span>
<span><span class="co">#&gt;     url = {https://github.com/microbiome/iSEEtree},</span></span>
<span><span class="co">#&gt;   }</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="background-knowledge">Background Knowledge<a class="anchor" aria-label="anchor" href="#background-knowledge"></a>
</h3>
<p>iSEEtree is based on many other packages and in particular on those
that have implemented the infrastructure needed for dealing with omics
data, microbiome data and interactive visualisation. That is, packages
like <a href="https://bioconductor.org/packages/3.18/bioc/html/SummarizedExperiment.html" class="external-link"><em>SummarizedExperiment</em></a>,
<a href="https://bioconductor.org/packages/3.18/bioc/html/TreeSummarizedExperiment.html" class="external-link"><em>TreeSummarizedExperiment</em></a>,
<a href="https://bioconductor.org/packages/3.18/bioc/html/mia.html" class="external-link"><em>mia</em></a>,
<a href="https://bioconductor.org/packages/3.18/bioc/html/iSEE.html" class="external-link"><em>iSEE</em></a>
and <a href="https://cran.r-project.org/web/packages/shiny/" class="external-link"><em>shiny</em></a>.</p>
<p>If you are asking yourself the question “Where do I start using
Bioconductor?” you might be interested in <a href="https://lcolladotor.github.io/2014/10/16/startbioc/#.VkOKbq6rRuU" class="external-link">this
blog post</a>.</p>
</div>
<div class="section level3">
<h3 id="help">Help<a class="anchor" aria-label="anchor" href="#help"></a>
</h3>
<p>As package developers, we try to explain clearly how to use our
packages and in which order to use the functions. But R and Bioconductor
have a steep learning curve so it is critical to learn where to ask for
help. The blog post quoted above mentions some but we would like to
highlight the <a href="https://support.bioconductor.org/" class="external-link">Bioconductor
support site</a> as the main resource for getting help: remember to use
the iSEEtree tag and check the <a href="https://support.bioconductor.org/t/iSEEtree/" class="external-link">older posts</a>.
Other alternatives are available such as creating GitHub issues and
tweeting. However, please note that if you want to receive help you
should adhere to the <a href="https://www.bioconductor.org/help/support/posting-guide/" class="external-link">posting
guidelines</a>. It is particularly critical that you provide a small
reproducible example and your session information so package developers
can track down the source of the error.</p>
</div>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>iSEEtree was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <a href="https://cran.r-project.org/web/packages/usethis/" class="external-link"><em>usethis</em></a>.</p>
<p>R session information:</p>
<pre><code><span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] scater_1.33.4                   ggplot2_3.5.1                   scuttle_1.15.4                 </span></span>
<span><span class="co">#&gt;  [4] mia_1.13.36                     MultiAssayExperiment_1.31.5     TreeSummarizedExperiment_2.13.0</span></span>
<span><span class="co">#&gt;  [7] Biostrings_2.73.2               XVector_0.45.0                  iSEEtree_0.99.7                </span></span>
<span><span class="co">#&gt; [10] iSEE_2.17.4                     SingleCellExperiment_1.27.2     SummarizedExperiment_1.35.2    </span></span>
<span><span class="co">#&gt; [13] Biobase_2.65.1                  GenomicRanges_1.57.1            GenomeInfoDb_1.41.1            </span></span>
<span><span class="co">#&gt; [16] IRanges_2.39.2                  S4Vectors_0.43.2                BiocGenerics_0.51.2            </span></span>
<span><span class="co">#&gt; [19] MatrixGenerics_1.17.0           matrixStats_1.4.1               RefManageR_1.4.0               </span></span>
<span><span class="co">#&gt; [22] BiocStyle_2.33.1               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] splines_4.4.1               later_1.3.2                 ggplotify_0.1.2             tibble_3.2.1               </span></span>
<span><span class="co">#&gt;   [5] polyclip_1.10-7             rpart_4.1.23                DirichletMultinomial_1.47.0 lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;   [9] doParallel_1.0.17           miaViz_1.13.10              lattice_0.22-6              MASS_7.3-61                </span></span>
<span><span class="co">#&gt;  [13] SnowballC_0.7.1             backports_1.5.0             magrittr_2.0.3              Hmisc_5.1-3                </span></span>
<span><span class="co">#&gt;  [17] sass_0.4.9                  rmarkdown_2.28              jquerylib_0.1.4             yaml_2.3.10                </span></span>
<span><span class="co">#&gt;  [21] httpuv_1.6.15               DBI_1.2.3                   minqa_1.2.8                 RColorBrewer_1.1-3         </span></span>
<span><span class="co">#&gt;  [25] lubridate_1.9.3             abind_1.4-8                 zlibbioc_1.51.1             purrr_1.0.2                </span></span>
<span><span class="co">#&gt;  [29] ggraph_2.2.1                yulab.utils_0.1.7           nnet_7.3-19                 tweenr_2.0.3               </span></span>
<span><span class="co">#&gt;  [33] sandwich_3.1-1              circlize_0.4.16             GenomeInfoDbData_1.2.13     ggrepel_0.9.6              </span></span>
<span><span class="co">#&gt;  [37] tokenizers_0.3.0            irlba_2.3.5.1               tidytree_0.4.6              vegan_2.6-8                </span></span>
<span><span class="co">#&gt;  [41] rbiom_1.0.3                 permute_0.9-7               pkgdown_2.1.1               DelayedMatrixStats_1.27.3  </span></span>
<span><span class="co">#&gt;  [45] codetools_0.2-20            DelayedArray_0.31.12        ggforce_0.4.2               DT_0.33                    </span></span>
<span><span class="co">#&gt;  [49] xml2_1.3.6                  tidyselect_1.2.1            shape_1.4.6.1               aplot_0.2.3                </span></span>
<span><span class="co">#&gt;  [53] farver_2.1.2                UCSC.utils_1.1.0            lme4_1.1-35.5               ScaledMatrix_1.13.0        </span></span>
<span><span class="co">#&gt;  [57] viridis_0.6.5               shinyWidgets_0.8.7          base64enc_0.1-3             jsonlite_1.8.9             </span></span>
<span><span class="co">#&gt;  [61] GetoptLong_1.0.5            BiocNeighbors_1.99.1        tidygraph_1.3.1             decontam_1.25.0            </span></span>
<span><span class="co">#&gt;  [65] Formula_1.2-5               iterators_1.0.14            systemfonts_1.1.0           foreach_1.5.2              </span></span>
<span><span class="co">#&gt;  [69] ggnewscale_0.5.0            tools_4.4.1                 treeio_1.29.1               ragg_1.3.3                 </span></span>
<span><span class="co">#&gt;  [73] Rcpp_1.0.13                 glue_1.8.0                  gridExtra_2.3               SparseArray_1.5.41         </span></span>
<span><span class="co">#&gt;  [77] xfun_0.47                   mgcv_1.9-1                  dplyr_1.1.4                 withr_3.0.1                </span></span>
<span><span class="co">#&gt;  [81] shinydashboard_0.7.2        BiocManager_1.30.25         fastmap_1.2.0               boot_1.3-31                </span></span>
<span><span class="co">#&gt;  [85] bluster_1.15.1              fansi_1.0.6                 shinyjs_2.1.0               digest_0.6.37              </span></span>
<span><span class="co">#&gt;  [89] rsvd_1.0.5                  gridGraphics_0.5-1          timechange_0.3.0            R6_2.5.1                   </span></span>
<span><span class="co">#&gt;  [93] mime_0.12                   textshaping_0.4.0           colorspace_2.1-1            listviewer_4.0.0           </span></span>
<span><span class="co">#&gt;  [97] lpSolve_5.6.21              utf8_1.2.4                  tidyr_1.3.1                 generics_0.1.3             </span></span>
<span><span class="co">#&gt; [101] data.table_1.16.0           DECIPHER_3.1.4              graphlayouts_1.2.0          httr_1.4.7                 </span></span>
<span><span class="co">#&gt; [105] htmlwidgets_1.6.4           S4Arrays_1.5.10             pkgconfig_2.0.3             gtable_0.3.5               </span></span>
<span><span class="co">#&gt; [109] ComplexHeatmap_2.21.1       janeaustenr_1.0.0           htmltools_0.5.8.1           bookdown_0.40              </span></span>
<span><span class="co">#&gt; [113] rintrojs_0.3.4              clue_0.3-65                 scales_1.3.0                png_0.1-8                  </span></span>
<span><span class="co">#&gt; [117] ggfun_0.1.6                 knitr_1.48                  rstudioapi_0.16.0           reshape2_1.4.4             </span></span>
<span><span class="co">#&gt; [121] rjson_0.2.23                nloptr_2.1.1                checkmate_2.3.2             nlme_3.1-166               </span></span>
<span><span class="co">#&gt; [125] shinyAce_0.4.2              zoo_1.8-12                  cachem_1.1.0                GlobalOptions_0.1.2        </span></span>
<span><span class="co">#&gt; [129] stringr_1.5.1               parallel_4.4.1              miniUI_0.1.1.1              vipor_0.4.7                </span></span>
<span><span class="co">#&gt; [133] foreign_0.8-87              desc_1.4.3                  pillar_1.9.0                grid_4.4.1                 </span></span>
<span><span class="co">#&gt; [137] vctrs_0.6.5                 slam_0.1-53                 promises_1.3.0              BiocSingular_1.21.4        </span></span>
<span><span class="co">#&gt; [141] beachmat_2.21.6             xtable_1.8-4                cluster_2.1.6               beeswarm_0.4.0             </span></span>
<span><span class="co">#&gt; [145] htmlTable_2.4.3             evaluate_1.0.0              mvtnorm_1.3-1               cli_3.6.3                  </span></span>
<span><span class="co">#&gt; [149] compiler_4.4.1              rlang_1.1.4                 crayon_1.5.3                tidytext_0.4.2             </span></span>
<span><span class="co">#&gt; [153] mediation_4.5.0             plyr_1.8.9                  fs_1.6.4                    ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt; [157] stringi_1.8.4               viridisLite_0.4.2           BiocParallel_1.39.0         munsell_0.5.1              </span></span>
<span><span class="co">#&gt; [161] lazyeval_0.2.2              colourpicker_1.3.0          Matrix_1.7-0                patchwork_1.3.0            </span></span>
<span><span class="co">#&gt; [165] sparseMatrixStats_1.17.2    shiny_1.9.1                 highr_0.11                  memoise_2.0.1              </span></span>
<span><span class="co">#&gt; [169] igraph_2.0.3                RcppParallel_5.1.9          bslib_0.8.0                 ggtree_3.13.1              </span></span>
<span><span class="co">#&gt; [173] bibtex_0.5.1                ape_5.8</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024) with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.
Citations were generated with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.28. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[2]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2024biocstyle"></a><a href="#cite-ole2024biocstyle">[3]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.33.1. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2024language"></a><a href="#cite-2024language">[4]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2024. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[5]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-xie2024knitr"></a><a href="#cite-xie2024knitr">[6]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.48. 2024. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Giulio Benedetti, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
