<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Metagenomic Data â€¢ iSEEtree</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Metagenomic Data">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEtree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.7</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/iSEEtree.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metagenomic_data.html">Metagenomic Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microbiome/iSEEtree/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Metagenomic Data</h1>
                        <h4 data-toc-skip class="author">Giulio
Benedetti</h4>
            <address class="author_afil">
      University of
Turku<br><a class="author_email" href="mailto:#"></a><a href="mailto:giulio.benedetti@utu.fi" class="email">giulio.benedetti@utu.fi</a>
      </address>
                  
            <h4 data-toc-skip class="date">2 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/microbiome/iSEEtree/blob/devel/vignettes/metagenomic_data.Rmd" class="external-link"><code>vignettes/metagenomic_data.Rmd</code></a></small>
      <div class="hidden name"><code>metagenomic_data.Rmd</code></div>

    </div>

    
    
<div class="sourceCode" id="section-cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Import packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/iSEEtree" class="external-link">iSEEtree</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: iSEE</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MatrixGenerics</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: matrixStats</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'MatrixGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span></span>
<span><span class="co">##     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span></span>
<span><span class="co">##     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span></span>
<span><span class="co">##     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span></span>
<span><span class="co">##     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span></span>
<span><span class="co">##     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span></span>
<span><span class="co">##     colWeightedMeans, colWeightedMedians, colWeightedSds,</span></span>
<span><span class="co">##     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span></span>
<span><span class="co">##     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span></span>
<span><span class="co">##     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span></span>
<span><span class="co">##     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span></span>
<span><span class="co">##     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span></span>
<span><span class="co">##     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span></span>
<span><span class="co">##     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span></span>
<span><span class="co">##     rowWeightedSds, rowWeightedVars</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomicRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: stats4</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: BiocGenerics</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'BiocGenerics'</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:stats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     IQR, mad, sd, var, xtabs</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyDuplicated, aperm, append, as.data.frame, basename, cbind,</span></span>
<span><span class="co">##     colnames, dirname, do.call, duplicated, eval, evalq, Filter, Find,</span></span>
<span><span class="co">##     get, grep, grepl, intersect, is.unsorted, lapply, Map, mapply,</span></span>
<span><span class="co">##     match, mget, order, paste, pmax, pmax.int, pmin, pmin.int,</span></span>
<span><span class="co">##     Position, rank, rbind, Reduce, rownames, sapply, saveRDS, setdiff,</span></span>
<span><span class="co">##     table, tapply, union, unique, unsplit, which.max, which.min</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: S4Vectors</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'S4Vectors'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:utils':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     findMatches</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     expand.grid, I, unname</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: IRanges</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: GenomeInfoDb</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biobase</span></span></code></pre>
<pre><code><span><span class="co">## Welcome to Bioconductor</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     Vignettes contain introductory material; view with</span></span>
<span><span class="co">##     'browseVignettes()'. To cite Bioconductor, see</span></span>
<span><span class="co">##     'citation("Biobase")', and for packages 'citation("pkgname")'.</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biobase'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:MatrixGenerics':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## The following objects are masked from 'package:matrixStats':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     anyMissing, rowMedians</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: SingleCellExperiment</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'iSEEtree'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:iSEE':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     iSEE</span></span></code></pre>
<div class="sourceCode" id="section-cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: TreeSummarizedExperiment</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: Biostrings</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: XVector</span></span></code></pre>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'Biostrings'</span></span></code></pre>
<pre><code><span><span class="co">## The following object is masked from 'package:base':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     strsplit</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: MultiAssayExperiment</span></span></code></pre>
<pre><code><span><span class="co">## This is mia version 1.13.36</span></span>
<span><span class="co">## - Online documentation and vignettes: https://microbiome.github.io/mia/</span></span>
<span><span class="co">## - Online book 'Orchestrating Microbiome Analysis (OMA)': https://microbiome.github.io/OMA/docs/devel/</span></span></code></pre>
<div class="sourceCode" id="section-cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/EBI-Metagenomics/MGnifyR" class="external-link">MGnifyR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Loading required package: scuttle</span></span></code></pre>
<pre><code><span><span class="co">## Loading required package: ggplot2</span></span></code></pre>
<div class="sourceCode" id="section-cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Instantiate mgnify client</span></span>
<span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MGnifyR/man/MgnifyClient.html" class="external-link">MgnifyClient</a></span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="section-cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch sample accession numbers</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MGnifyR/man/searchAnalysis.html" class="external-link">searchAnalysis</a></span><span class="op">(</span><span class="va">mg</span>,</span>
<span>                          accession <span class="op">=</span> <span class="st">"MGYS00002504"</span>,</span>
<span>                          type <span class="op">=</span> <span class="st">"studies"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fetching analyses...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span></span></code></pre>
<div class="sourceCode" id="section-cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch samples as TreeSE</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/MGnifyR/man/getResult.html" class="external-link">getResult</a></span><span class="op">(</span><span class="va">mg</span>, <span class="va">samples</span>,</span>
<span>                 get.taxa <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 get.func <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Fetching taxonomy data...</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=========                                                             |  12%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |==================                                                    |  25%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |==========================                                            |  38%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |===================================                                   |  50%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |============================================                          |  62%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |====================================================                  |  75%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |=============================================================         |  88%</span></span></code></pre>
<pre><code><span><span class="co">## Warning in loadFromBiom(biom_path, removeTaxaPrefixes = TRUE, only.taxa.col =</span></span>
<span><span class="co">## TRUE, : 'loadFromBiom' is deprecated. Use 'importBIOM' instead.</span></span></code></pre>
<pre><code><span><span class="co">##   |                                                                              |======================================================================| 100%</span></span></code></pre>
<pre><code><span><span class="co">## Merging with full join...</span></span></code></pre>
<pre><code><span><span class="co">## 1/82/83/84/85/86/87/88/8</span></span></code></pre>
<div class="sourceCode" id="section-cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Transform counts to relabundance</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>                      assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Transform relabundance to CLR</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"clr"</span>,</span>
<span>                      assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>                      pseudocount <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## A pseudocount of 1.21921482565228e-05 was applied.</span></span></code></pre>
<div class="sourceCode" id="section-cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Standardise CLR</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"standardize"</span>,</span>
<span>                      assay.type <span class="op">=</span> <span class="st">"clr"</span>,</span>
<span>                      MARGIN <span class="op">=</span> <span class="st">"features"</span>,</span>
<span>                      name <span class="op">=</span> <span class="st">"clr-z"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="section-cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate alpha diversity</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/addAlpha.html" class="external-link">addAlpha</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coverage_diversity"</span>, <span class="st">"shannon_diversity"</span><span class="op">)</span>,</span>
<span>                assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Estimate beta diversity</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>              FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>,</span>
<span>              method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>              assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="section-cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove long variables</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">$</span><span class="va">study_attributes.study.abstract</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">$</span><span class="va">sample_sample.desc</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></code></pre></div>
<div class="sourceCode" id="section-cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Launch app</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/iSEE.html">iSEE</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Giulio Benedetti, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
