<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Metagenomic Data • iSEEtree</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Metagenomic Data">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEtree</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/iSEEtree.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/metagenomic_data.html">Metagenomic Data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microbiome/iSEEtree/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Metagenomic Data</h1>
                        <h4 data-toc-skip class="author">Giulio
Benedetti</h4>
            <address class="author_afil">
      University of
Turku<br><a class="author_email" href="mailto:#"></a><a href="mailto:giulio.benedetti@utu.fi" class="email">giulio.benedetti@utu.fi</a>
      </address>
                  
            <h4 data-toc-skip class="date">16 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/microbiome/iSEEtree/blob/coltree&amp;loadings/vignettes/metagenomic_data.Rmd" class="external-link"><code>vignettes/metagenomic_data.Rmd</code></a></small>
      <div class="hidden name"><code>metagenomic_data.Rmd</code></div>

    </div>

    
    
<p>Here, we demonstrate how to fetch a metagenomic dataset from the <a href="https://www.ebi.ac.uk/about/teams/microbiome-informatics/mgnify/" class="external-link">MGnify
database</a> and explore it with the help of iSEEtree. For this purpose,
we will use a relatively large dataset obtained by shotgun metagenomic
sequencing of water samples from the Arctic Ocean during winter-spring
transition <span class="citation">(Sousa et al. 2019)</span>.</p>
<p>First, the following packages are imported to run the analysis.</p>
<p>Then, the Mgnify client is instantiated.</p>
<div class="sourceCode" id="section-cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Instantiate mgnify client</span></span>
<span><span class="va">mg</span> <span class="op">&lt;-</span> <span class="fu">MgnifyClient</span><span class="op">(</span>useCache <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>We search all samples from the study of interest, which has the
accession MGYS00001869.</p>
<div class="sourceCode" id="section-cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch sample accession numbers</span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu">searchAnalysis</span><span class="op">(</span><span class="va">mg</span>,</span>
<span>                          accession <span class="op">=</span> <span class="st">"MGYS00001869"</span>,</span>
<span>                          type <span class="op">=</span> <span class="st">"studies"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span></span></code></pre>
<p>The samples are then fetched and combined into a TreeSE. This step
might take upto 10 minutes.</p>
<div class="sourceCode" id="section-cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Fetch samples as TreeSE</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">getResult</span><span class="op">(</span><span class="va">mg</span>, <span class="va">samples</span>,</span>
<span>                 get.taxa <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                 get.func <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt;   |                                                                              |                                                                      |   0%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |===                                                                   |   4%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=====                                                                 |   7%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |========                                                              |  11%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |==========                                                            |  15%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=============                                                         |  19%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |================                                                      |  22%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |==================                                                    |  26%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=====================                                                 |  30%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=======================                                               |  33%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |==========================                                            |  37%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=============================                                         |  41%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |===============================                                       |  44%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |==================================                                    |  48%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |====================================                                  |  52%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=======================================                               |  56%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=========================================                             |  59%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |============================================                          |  63%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |===============================================                       |  67%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=================================================                     |  70%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |====================================================                  |  74%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |======================================================                |  78%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=========================================================             |  81%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |============================================================          |  85%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |==============================================================        |  89%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |=================================================================     |  93%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |===================================================================   |  96%</span></span></code></pre>
<pre><code><span><span class="co">#&gt;   |                                                                              |======================================================================| 100%</span></span></code></pre>
<div class="sourceCode" id="section-cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># View TreeSE</span></span>
<span><span class="va">tse</span></span></code></pre></div>
<pre><code><span><span class="co">#&gt; class: TreeSummarizedExperiment </span></span>
<span><span class="co">#&gt; dim: 4057 27 </span></span>
<span><span class="co">#&gt; metadata(0):</span></span>
<span><span class="co">#&gt; assays(1): counts</span></span>
<span><span class="co">#&gt; rownames(4057): 384552 1130548 ... 41722 5820</span></span>
<span><span class="co">#&gt; rowData names(9): Superkingdom Kingdom ... Species taxonomy_unparsed</span></span>
<span><span class="co">#&gt; colnames(27): MGYA00132764 MGYA00132765 ... MGYA00579273 MGYA00579274</span></span>
<span><span class="co">#&gt; colData names(64): analysis_pipeline.version analysis_accession ...</span></span>
<span><span class="co">#&gt;   run_accession biome_string</span></span>
<span><span class="co">#&gt; reducedDimNames(0):</span></span>
<span><span class="co">#&gt; mainExpName: NULL</span></span>
<span><span class="co">#&gt; altExpNames(0):</span></span>
<span><span class="co">#&gt; rowLinks: NULL</span></span>
<span><span class="co">#&gt; rowTree: NULL</span></span>
<span><span class="co">#&gt; colLinks: NULL</span></span>
<span><span class="co">#&gt; colTree: NULL</span></span></code></pre>
<p>We apply different transformations to the counts assay, as certain
graphs provide more useful information and more aesthetically appealing
visualisation when created with, for example, relative abundance or
standardised assays.</p>
<div class="sourceCode" id="section-cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Transform counts to relabundance</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>                      assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Transform relabundance to CLR</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"clr"</span>,</span>
<span>                      assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>                      pseudocount <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Standardise CLR</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/transformAssay.html" class="external-link">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                      method <span class="op">=</span> <span class="st">"standardize"</span>,</span>
<span>                      assay.type <span class="op">=</span> <span class="st">"clr"</span>,</span>
<span>                      MARGIN <span class="op">=</span> <span class="st">"features"</span>,</span>
<span>                      name <span class="op">=</span> <span class="st">"clr-z"</span><span class="op">)</span></span></code></pre></div>
<p>We estimate information on alpha and beta diversity so that it can be
accessed from the app.</p>
<div class="sourceCode" id="section-cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimate alpha diversity</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/mia/man/addAlpha.html" class="external-link">addAlpha</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coverage_diversity"</span>, <span class="st">"shannon_diversity"</span><span class="op">)</span>,</span>
<span>                assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Estimate beta diversity</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>              FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>,</span>
<span>              method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>              ncomp <span class="op">=</span> <span class="fl">10</span>,</span>
<span>              assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></code></pre></div>
<p>To prevent the column data table to take too much space, we remove
the two long variables that are causing the expansion. In principle, the
app can also run without this deletion.</p>
<div class="sourceCode" id="section-cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Remove long variables</span></span>
<span><span class="va">max_col_length</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span>, <span class="fl">2</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nchar.html" class="external-link">nchar</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span> , <span class="va">max_col_length</span> <span class="op">&lt;</span> <span class="fl">20</span> <span class="op">|</span> <span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">max_col_length</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>Finally, the app is launched by passing the TreeSE object to
<code>iSEE</code>. After tuning the parameters for the different panels,
the app appears as illustrated by the image below.</p>
<div class="sourceCode" id="section-cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Launch app</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/iSEE.html">iSEE</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="screenshots%2Fmetagenomic_data.png" width="100%"></p>
<p>To know more about how to explore big data with iSEE and iSEEtree,
check the related <a href="https://isee.github.io/iSEE/articles/bigdata.html" class="external-link">iSEE
article</a>.</p>
<pre><code><span><span class="co">#&gt; R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; other attached packages:</span></span>
<span><span class="co">#&gt;  [1] scater_1.33.4                   ggplot2_3.5.1                   scuttle_1.15.4                 </span></span>
<span><span class="co">#&gt;  [4] MGnifyR_1.1.0                   mia_1.13.46                     TreeSummarizedExperiment_2.13.0</span></span>
<span><span class="co">#&gt;  [7] Biostrings_2.73.2               XVector_0.45.0                  MultiAssayExperiment_1.31.5    </span></span>
<span><span class="co">#&gt; [10] iSEEtree_0.99.8                 iSEE_2.17.4                     SingleCellExperiment_1.27.2    </span></span>
<span><span class="co">#&gt; [13] SummarizedExperiment_1.35.4     Biobase_2.65.1                  GenomicRanges_1.57.2           </span></span>
<span><span class="co">#&gt; [16] GenomeInfoDb_1.41.2             IRanges_2.39.2                  S4Vectors_0.43.2               </span></span>
<span><span class="co">#&gt; [19] BiocGenerics_0.51.3             MatrixGenerics_1.17.0           matrixStats_1.4.1              </span></span>
<span><span class="co">#&gt; [22] BiocStyle_2.33.1               </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;   [1] fs_1.6.4                    tidyjson_0.3.2              DirichletMultinomial_1.47.0 httr_1.4.7                 </span></span>
<span><span class="co">#&gt;   [5] RColorBrewer_1.1-3          doParallel_1.0.17           tools_4.4.1                 backports_1.5.0            </span></span>
<span><span class="co">#&gt;   [9] utf8_1.2.4                  R6_2.5.1                    DT_0.33                     vegan_2.6-8                </span></span>
<span><span class="co">#&gt;  [13] lazyeval_0.2.2              mgcv_1.9-1                  rhdf5filters_1.17.0         GetoptLong_1.0.5           </span></span>
<span><span class="co">#&gt;  [17] permute_0.9-7               withr_3.0.1                 gridExtra_2.3               cli_3.6.3                  </span></span>
<span><span class="co">#&gt;  [21] textshaping_0.4.0           shinyjs_2.1.0               sandwich_3.1-1              slam_0.1-54                </span></span>
<span><span class="co">#&gt;  [25] sass_0.4.9                  mvtnorm_1.3-1               pkgdown_2.1.1               systemfonts_1.1.0          </span></span>
<span><span class="co">#&gt;  [29] yulab.utils_0.1.7           foreign_0.8-87              decontam_1.25.0             rstudioapi_0.16.0          </span></span>
<span><span class="co">#&gt;  [33] generics_0.1.3              gridGraphics_0.5-1          shape_1.4.6.1               dplyr_1.1.4                </span></span>
<span><span class="co">#&gt;  [37] rbiom_1.0.3                 Matrix_1.7-0                biomformat_1.33.0           ggbeeswarm_0.7.2           </span></span>
<span><span class="co">#&gt;  [41] fansi_1.0.6                 DECIPHER_3.1.4              abind_1.4-8                 lifecycle_1.0.4            </span></span>
<span><span class="co">#&gt;  [45] yaml_2.3.10                 rhdf5_2.49.0                SparseArray_1.5.44          grid_4.4.1                 </span></span>
<span><span class="co">#&gt;  [49] promises_1.3.0              crayon_1.5.3                shinydashboard_0.7.2        miniUI_0.1.1.1             </span></span>
<span><span class="co">#&gt;  [53] lattice_0.22-6              beachmat_2.21.6             pillar_1.9.0                knitr_1.48                 </span></span>
<span><span class="co">#&gt;  [57] ComplexHeatmap_2.21.1       rjson_0.2.23                boot_1.3-31                 lpSolve_5.6.21             </span></span>
<span><span class="co">#&gt;  [61] codetools_0.2-20            glue_1.8.0                  ggfun_0.1.6                 data.table_1.16.2          </span></span>
<span><span class="co">#&gt;  [65] vctrs_0.6.5                 png_0.1-8                   treeio_1.29.1               urltools_1.7.3             </span></span>
<span><span class="co">#&gt;  [69] gtable_0.3.5                assertthat_0.2.1            cachem_1.1.0                xfun_0.48                  </span></span>
<span><span class="co">#&gt;  [73] S4Arrays_1.5.11             mime_0.12                   tidygraph_1.3.1             iterators_1.0.14           </span></span>
<span><span class="co">#&gt;  [77] bluster_1.15.1              nlme_3.1-166                ggtree_3.13.1               SnowballC_0.7.1            </span></span>
<span><span class="co">#&gt;  [81] bslib_0.8.0                 irlba_2.3.5.1               vipor_0.4.7                 rpart_4.1.23               </span></span>
<span><span class="co">#&gt;  [85] colorspace_2.1-1            DBI_1.2.3                   Hmisc_5.1-3                 nnet_7.3-19                </span></span>
<span><span class="co">#&gt;  [89] tidyselect_1.2.1            compiler_4.4.1              curl_5.2.3                  htmlTable_2.4.3            </span></span>
<span><span class="co">#&gt;  [93] BiocNeighbors_1.99.2        desc_1.4.3                  DelayedArray_0.31.14        colourpicker_1.3.0         </span></span>
<span><span class="co">#&gt;  [97] bookdown_0.41               triebeard_0.4.1             checkmate_2.3.2             scales_1.3.0               </span></span>
<span><span class="co">#&gt; [101] stringr_1.5.1               digest_0.6.37               minqa_1.2.8                 rmarkdown_2.28             </span></span>
<span><span class="co">#&gt; [105] miaViz_1.13.14              htmltools_0.5.8.1           pkgconfig_2.0.3             base64enc_0.1-3            </span></span>
<span><span class="co">#&gt; [109] lme4_1.1-35.5               sparseMatrixStats_1.17.2    highr_0.11                  listviewer_4.0.0           </span></span>
<span><span class="co">#&gt; [113] fastmap_1.2.0               rlang_1.1.4                 GlobalOptions_0.1.2         htmlwidgets_1.6.4          </span></span>
<span><span class="co">#&gt; [117] UCSC.utils_1.1.0            shiny_1.9.1                 DelayedMatrixStats_1.27.3   farver_2.1.2               </span></span>
<span><span class="co">#&gt; [121] jquerylib_0.1.4             zoo_1.8-12                  jsonlite_1.8.9              BiocParallel_1.39.0        </span></span>
<span><span class="co">#&gt; [125] tokenizers_0.3.0            BiocSingular_1.21.4         magrittr_2.0.3              Formula_1.2-5              </span></span>
<span><span class="co">#&gt; [129] GenomeInfoDbData_1.2.13     ggplotify_0.1.2             patchwork_1.3.0             Rhdf5lib_1.27.0            </span></span>
<span><span class="co">#&gt; [133] munsell_0.5.1               Rcpp_1.0.13                 ape_5.8                     ggnewscale_0.5.0           </span></span>
<span><span class="co">#&gt; [137] viridis_0.6.5               stringi_1.8.4               rintrojs_0.3.4              ggraph_2.2.1               </span></span>
<span><span class="co">#&gt; [141] zlibbioc_1.51.1             MASS_7.3-61                 plyr_1.8.9                  mediation_4.5.0            </span></span>
<span><span class="co">#&gt; [145] parallel_4.4.1              ggrepel_0.9.6               graphlayouts_1.2.0          splines_4.4.1              </span></span>
<span><span class="co">#&gt; [149] circlize_0.4.16             igraph_2.0.3                reshape2_1.4.4              ScaledMatrix_1.13.0        </span></span>
<span><span class="co">#&gt; [153] evaluate_1.0.1              tidytext_0.4.2              RcppParallel_5.1.9          BiocManager_1.30.25        </span></span>
<span><span class="co">#&gt; [157] nloptr_2.1.1                foreach_1.5.2               tweenr_2.0.3                httpuv_1.6.15              </span></span>
<span><span class="co">#&gt; [161] tidyr_1.3.1                 purrr_1.0.2                 polyclip_1.10-7             clue_0.3-65                </span></span>
<span><span class="co">#&gt; [165] ggforce_0.4.2               rsvd_1.0.5                  xtable_1.8-4                tidytree_0.4.6             </span></span>
<span><span class="co">#&gt; [169] janeaustenr_1.0.0           later_1.3.2                 viridisLite_0.4.2           ragg_1.3.3                 </span></span>
<span><span class="co">#&gt; [173] tibble_3.2.1                aplot_0.2.3                 memoise_2.0.1               beeswarm_0.4.0             </span></span>
<span><span class="co">#&gt; [177] cluster_2.1.6               shinyWidgets_0.8.7          shinyAce_0.4.2</span></span></code></pre>
<div id="section-refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="section-ref-de2019diversity" class="csl-entry">
Sousa, António Gaspar G de, Maria Paola Tomasino, Pedro Duarte, Mar
Fernández-Méndez, Philipp Assmy, Hugo Ribeiro, Jaroslaw Surkont, et al.
2019. <span>“Diversity and Composition of Pelagic Prokaryotic and
Protist Communities in a Thin Arctic Sea-Ice Regime.”</span>
<em>Microbial Ecology</em> 78: 388–408. <a href="https://doi.org/10.1007/s00248-018-01314-2" class="external-link">https://doi.org/10.1007/s00248-018-01314-2</a>.
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Giulio Benedetti, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
